File: Easy/JavaScript/404. Sum of Left Leaves.js
var sumOfLeftLeaves = function(root) {
    if(root == null || (root.left == null && root.right == null))
        return 0;
    
    let sum = 0;
    if(root.left != null && root.left.left == null && root.left.right == null)
        sum += root.left.val;
    
    return sum + sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
};

File: Easy/Python/404. Sum of Left Leaves.py
class Solution(object):
    def sumOfLeftLeaves(self, root):
        if not root or (not root.left and not root.right):
            return 0
        
        this_sum = 0
        if root.left and not root.left.left and not root.left.right:
            this_sum += root.left.val
        
        return this_sum + self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right)
